seqr/utils/social_auth_pipeline.py

```diff
@@ -13,10 +13,13 @@ def validate_anvil_registration(backend, response, *args, **kwargs):
             anvil_call('get', 'register', response['access_token'])
         except TerraNotFoundException as et:
             logger.warning('User {} is trying to login without registration on AnVIL. {}'.format(response['email'], str(et)))
-            return redirect('/login?googleLoginFailed=true')
+            return redirect('/login?anvilLoginFailed=true')
 
 
-def log_signed_in(backend, response, is_new=False, *args, **kwargs):
+def log_signed_in(backend, response, user=None, is_new=False, *args, **kwargs):
+    if not user:
+        logger.warning('Google user {} is trying to login without an existing seqr account'.format(response['email']))
+        return redirect('/login?googleLoginFailed=true')
```

Comment:
this is important logic, but it doesn't belong in the `log_signed_in` helper step at the end of the pipeline. You should add a custom step named something like `validate_existing_user` that is run as soon in the pipeline as possible and is only in the non-anvil pipeline